### PR TITLE
[SPARK-14828][SQL] Start SparkSession in REPL instead of SQLContext

### PR DESCRIPTION
**Pull request by andrewor14**
## What changes were proposed in this pull request?

```
Spark context available as 'sc' (master = local[*], app id = local-1461283768192).
Spark session available as 'spark'.
Welcome to
      ____              __
     / __/__  ___ _____/ /__
    _\ \/ _ \/ _ `/ __/  '_/
   /___/ .__/\_,_/_/ /_/\_\   version 2.0.0-SNAPSHOT
      /_/

Using Scala version 2.11.8 (Java HotSpot(TM) 64-Bit Server VM, Java 1.7.0_51)
Type in expressions to have them evaluated.
Type :help for more information.

scala> sql("SHOW TABLES").collect()
16/04/21 17:09:39 WARN ObjectStore: Version information not found in metastore. hive.metastore.schema.verification is not enabled so recording the schema version 1.2.0
16/04/21 17:09:39 WARN ObjectStore: Failed to get database default, returning NoSuchObjectException
res0: Array[org.apache.spark.sql.Row] = Array([src,false])

scala> sql("SHOW TABLES").collect()
res1: Array[org.apache.spark.sql.Row] = Array([src,false])

scala> spark.createDataFrame(Seq((1, 1), (2, 2), (3, 3)))
res2: org.apache.spark.sql.DataFrame = [_1: int, _2: int]
```

Hive things are loaded lazily.
## How was this patch tested?

Manual.

### CODE REVIEW

**Comment by rxin** [comment]
I think there was a suggestion from @marmbrus to just name this "spark". Would it cause any problems because it conflicts with the package name?

**Comment by andrewor14** [comment]
OK, I like spark better too. Let's try it out.

**Comment by marmbrus** [comment]
Yeah, if there are problems we don't have to do that, but `val df = spark.read.json(...)` is pretty nice

**Comment by andrewor14** [comment]
I don't see any problems off the top of my head. The only collision is when people do `import org.apache.spark`, but who does that?

**Comment by rxin** [comment]
LGTM - let's see if we run into issues in the future with this.

File: repl/scala-2.10/src/main/scala/org/apache/spark/repl/SparkILoop.scala [hunk 1]
```diff
@@ -1026,21 +1025,7 @@ class SparkILoop(
...
-      logInfo("Created sql context (with Hive support)..")
-    }
-    catch {
-      case _: java.lang.ClassNotFoundException | _: java.lang.NoClassDefFoundError =>
-        sqlContext = new SQLContext(sparkContext)
-        logInfo("Created sql context..")
-    }
-    sqlContext
-  }
+  def createSparkSession(): SparkSession = Main.createSparkSession()
```
Comment (yhuai) [inline repl/scala-2.10/src/main/scala/org/apache/spark/repl/SparkILoop.scala]: Does scala 2.10's Main have `createSparkSession`?

Comment (andrewor14) [inline repl/scala-2.10/src/main/scala/org/apache/spark/repl/SparkILoop.scala]: hm, is there a way to do this without duplicating code?

Comment (yhuai) [inline repl/scala-2.10/src/main/scala/org/apache/spark/repl/SparkILoop.scala]: Not very sure. How about we still duplicate the code for now?

**Comment by rxin** [comment]
Thanks - merging in master.

**Comment by felixcheung** [comment]
should this go to pyspark and sparkR shell as well?
or rather, should SparkSession be in Python and R first?

**Comment by andrewor14** [comment]
it'll be there later

**Pull request closed without merging**